File: function-aws/features/test-aws-junit/skeleton/src/test/java/@defaultPackage.path@/@project.className@FunctionTest.java
package @defaultPackage@;

import io.micronaut.test.annotation.MicronautTest;
import org.junit.jupiter.api.Test;
import javax.inject.Inject;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
public class @project.className@FunctionTest {

    @Inject
    @project.className@Client client;

    @Test
    public void testFunction() throws Exception {
    	@project.className@ body = new @project.className@();
    	body.setName("@project.name@");
        assertEquals("@project.name@", client.apply(body).blockingGet().getName());
    }
}

File: function-aws/features/test-aws-spek/skeleton/src/test/kotlin/@defaultPackage.path@/@project.className@FunctionTest.kt
package @defaultPackage@

import io.micronaut.context.ApplicationContext
import io.micronaut.runtime.server.EmbeddedServer
import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it
import org.junit.jupiter.api.Assertions.assertEquals


class @project.className@FunctionTest: Spek({

    describe("@project.name@ function") {
        val server = ApplicationContext.run(EmbeddedServer::class.java)
        val client = server.applicationContext.getBean(@project.className@Client::class.java)

        it("should return '@project.name@'") {
            val body = @project.className@()
            body.name = "@project.name@"
            assertEquals(
                "@project.name@", 
                client.apply(body).blockingGet().name
            )
        }

        afterGroup {
            server.stop()
        }
    }
})
